Fixed menus

TBubba opened the pull request:
Added a lot of comments in the singleplayer and multiplayer menu.
Enhanced (and fix some bugs) in the server browser.
Created a browser for singleplayer games (browse through saved games) based on the multiplayer browser.
Sorted out some of the messy parts of the code.

(First time ever pull requesting, so I'm not entirely sure what to type here or what a pull request actually is)

croxxx commented:
Hm, seems like you did quite some stuff. Nice to see :smile: 

A few points I've come across:
- Singleplayer save management does not work. Produces a crash for me.
- You changed the solution file and build commands. Why so? They work perfectly for me (on Windows as well as Linux)
- It seems you fixed a build error that I already fixed earlier today :)
- You reformatted quite a lot of if-statements:
```csharp
if (cond)
    stuff();
```
instead of
```csharp
if (cond)
{
    stuff();
}
```
This isn't particularly bad but should be consistent throughout the application.

---
Overall some good work.

TBubba commented:
* I found out that the singleplayer menu crashes if the user has no saved games, I'll fix that right away.
* The solution commands gave me an error (error code 9009 if I'm not mistaken), but I switched to the previous one now and it works fine. Not sure why.
* ^^
* I'm not entirely sure how I should write my code. Should I continue without the brackets or should I use them?

Expect a commit very soon. :blush:

TBubba commented:
Apparently the __page count__ could be negative in certain resolutions (and cause a crash), but that is probably not what happened to you. I would assume that Platform.CreateSavegamesDirectory() could cause crashes, I'll look into it.

croxxx commented:
CiTo is a bit picky as it doesn't support the full C# syntax. The build commands should definitely work, though...
No, Travis failing is a result of it trying to upload the build which it isn't allowed from repos other than main repo. I'll try to get this fixed as it can be kinda annoying.

croxxx commented:
On brackets:
In most of the code I've seen the if-statements with brackets always included. I think this should stay that way.

Oh, and I *think* you can remove/exclude single commits from pull requests (just noticed the solution file changes are still in the request).

TBubba commented:
I found a couple of bugs and fixed them during the day, including the one you were suffering from. The menus are as far as I know crash-proof now. :smile:
There are however some more things I would like to improve (especially the refresh button I added to the singleplayer menu) before insisting on a merge.

Regarding the brackets. I'm switching back to using brackets like the majority of the code. But would it be okay for me to not use brackets in switches or loops where I do something like the following?
    if (someIndex == -1) break;
and
    if (someValue == null) continue;
(I saw something like this somewhere else in the project if I'm not mistaken)

I'll switch back to the previous project file (but with the new class) in the upcoming commit.

croxxx commented:
Hm, I think so far this version has been used:
```csharp
if (cond) { code(); }
```

TBubba commented:
That seems more likely, I'll start typing like that instead.

TBubba commented:
I would like to add a _Remove_ button in the singleplayer menu for removing saved games, but it seems like there is no way to do that right now.
Should I add a method for removing files in _Platform_?

I would also like to make double-clicking servers (or saves) a way to connect to (or open) them.
I'm thinking of implementing a timer (counting the parameter from _Render(float dt)_ each frame) that resets whenever a widget is pressed, and then just check if the widget is pressed again before _x_ amount of time has passed. (This is mostly a note so I won't forget it before I implement it)

Also, I will be gone for a couple of days (but I'll still probably do some coding :steam_locomotive:).

colin500 commented:
can someone make a thing where once your signed in you dont have to wait a couple secs on the login page when your sign in

TBubba commented:
@Colin500 You wait at

TBubba commented:
I misclicked and did some crazy stuff (damn touch keyboard).

croxxx commented:
@colin500 I guess what you are experiencing is the actual login process.
Whenever you join a server you are logged in and depending on your PC and internet connection this may take a second.

colin500 commented:
ok

croxxx commented:
@TBubba What's the current status of this?

TBubba commented:
@croxxx I haven't touched it for a couple of weeks (school you know), but I'll get onto finishing it early July.
